scripts/xtensa-build-zephyr.py

```diff
@@ -0,0 +1,382 @@
+	zephyr_dir = pathlib.Path(WEST_TOP, "zephyr")
+	execute_command(f"git clone {args.url} {zephyr_dir}", check=True, timeout=1200)
+	execute_command(f"git fetch origin {args.z}", check=True, timeout=300, cwd=zephyr_dir)
+	execute_command(f"git checkout FETCH_HEAD", check=True, cwd=zephyr_dir)
+	execute_command(f"git -C {zephyr_dir} --no-pager log --oneline --graph --decorate"
+		" --max-count=20", check=True)
+	execute_command(f"west init -l {zephyr_dir}", check=True)
+	create_zephyr_sof_symlink()
+	execute_command("west update zephyr hal_xtensa", check=True, timeout=300, cwd=WEST_TOP)
+
+def build_platforms():
+	global WEST_TOP, SOF_TOP
+	# double check that west workspace is initialized
+	execute_command("west topdir", check=True, stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL,
+		cwd=WEST_TOP)
+	print(f"SOF_TOP={SOF_TOP}")
+	print(f"WEST_TOP={WEST_TOP}")
+	global STAGING_DIR
+	STAGING_DIR = pathlib.Path(WEST_TOP, "build-sof-staging", "sof")
+	# smex does not use 'install -D'
+	STAGING_DIR.mkdir(mode=511, parents=True, exist_ok=True)
+	for platform in args.platforms:
+		platform_dict = next(x for x in platform_list if x["name"] == platform)
+		XTENSA_TOOLS_ROOT = os.getenv("XTENSA_TOOLS_ROOT")
+		if XTENSA_TOOLS_ROOT and "XTENSA_TOOLS_VERSION" in platform_dict:
+			# set variables expected by zephyr/cmake/toolchain/xcc/generic.cmake
+			os.environ["ZEPHYR_TOOLCHAIN_VARIANT"] = "xcc"
+			XTENSA_TOOLCHAIN_PATH = str(pathlib.Path(XTENSA_TOOLS_ROOT, "install",
+				"tools").absolute())
+			os.environ["XTENSA_TOOLCHAIN_PATH"] = XTENSA_TOOLCHAIN_PATH
+			TOOLCHAIN_VER = platform_dict["XTENSA_TOOLS_VERSION"]
+			XTENSA_CORE = platform_dict["XTENSA_CORE"]
+			os.environ["TOOLCHAIN_VER"] = TOOLCHAIN_VER
+			print(f"XTENSA_TOOLCHAIN_PATH={XTENSA_TOOLCHAIN_PATH}")
+			print(f"TOOLCHAIN_VER={TOOLCHAIN_VER}")
+
+			# set variables expected by xcc toolchain
+			XTENSA_BUILDS_DIR=str(pathlib.Path(XTENSA_TOOLS_ROOT, "install", "builds",
+				TOOLCHAIN_VER).absolute())
+			XTENSA_SYSTEM = str(pathlib.Path(XTENSA_BUILDS_DIR, XTENSA_CORE, "config").absolute())
+			os.environ["XTENSA_SYSTEM"] = XTENSA_SYSTEM
+			print(f"XTENSA_SYSTEM={XTENSA_SYSTEM}")
+
+		platform_build_dir = f"build-{platform}"
+		west_verbose = "-v" if args.verbose_build else ""
+		PLAT_CONFIG = platform_dict["PLAT_CONFIG"]
+		cmake_args = f"-- {args.cmake_args}" if args.cmake_args else ""
+		build_cmd = f"west {west_verbose} build --build-dir {platform_build_dir}"
+		build_cmd += f" --board {PLAT_CONFIG} zephyr/samples/subsys/audio/sof {cmake_args}"
+		# Build
+		execute_command(build_cmd, check=True, cwd=WEST_TOP)
+		smex_executable = pathlib.Path(WEST_TOP, platform_build_dir, "zephyr", "smex_ep", "build",
+			"smex")
+		fw_ldc_file = pathlib.Path(STAGING_DIR, f"sof-{platform}.ldc")
+		input_elf_file = pathlib.Path(WEST_TOP, platform_build_dir, "zephyr", "zephyr.elf")
+		# Extract metadata
+		execute_command(f"{smex_executable} -l {fw_ldc_file} {input_elf_file}", check=True)
```

Comment:
I don't see how these can work with `shell=False`... should be:
```suggestion
		execute_command([smex_executable, "-l", fw_ldc_file, ,input_elf_file], check=True)
```

Reply:
It works because subprocess.run accepts a list of arguments or a single string separated with spaces.
Internally they are split into the list anyway.

Reply:
Seems that it is true only for Windows OS. Rewritten all execute_command arguments to lists.